tests/tests_unit/test_data_classes/test_assets.py

```diff
@@ -79,19 +79,22 @@ def test_get_files(self):
         "resource_class, resource_list_class, method",
         [(FileMetadata, FileMetadataList, "files"), (Event, EventList, "events")],
     )
-    @mock.patch("cognite.client.utils._concurrency")
-    def test_get_related_resources_should_not_return_duplicates(
-        self, mock_concurrency, resource_class, resource_list_class, method
-    ):
-        assets = AssetList([Asset(id=1), Asset(id=2), Asset(id=3)], cognite_client=mock.MagicMock())
+    def test_get_related_resources_should_not_return_duplicates(self, resource_class, resource_list_class, method):
         r1 = resource_class(id=1)
         r2 = resource_class(id=2)
         r3 = resource_class(id=3)
         resources_a1 = resource_list_class([r1])
         resources_a2 = resource_list_class([r2, r3])
         resources_a3 = resource_list_class([r2, r3])
 
-        mock_concurrency.execute_tasks_concurrently.return_value.results = [resources_a1, resources_a2, resources_a3]
+        mock_cognite_client = mock.MagicMock()
+        mock_method = getattr(mock_cognite_client, method)
+        mock_method.list.side_effect = [resources_a1, resources_a2, resources_a3]
+        mock_method._config = mock.Mock(max_workers=3)
```

Comment:
Is this necessary?

Reply:
It's necessary in order for the comparison on line 108 of _concurrency.py to work. The exact value of 3 is not important, though.